mosquitto/rootfs/usr/share/tempio/nginx.gtpl

```diff
@@ -35,6 +35,7 @@ http {
 
     location /authentication {
       proxy_set_header        X-Supervisor-Token "{{ env "SUPERVISOR_TOKEN" }}";
+      proxy_set_header        Authorization "";
```

Comment:
This is incorrect, this will break HA authentication.

Reply:
@frenck can you please elaborate how?

As far as the code in code when this was developed, the code checked first the Basic Auth, then the parameters that were passed in the URL.

HA uses Mosquitto Password and ACL files and doesn't authenticate via HTTP from this addon afaik.

Reply:
@frenck wondering if you saw my reply to you last response?

Reply:
@bytespider I missed that indeed. Sorry about that.

> and doesn't authenticate via HTTP from this addon afaik.

It does :) It authenticates against Home Assistant with Home Assistant user accounts.

Reply:
Okay, thank you.

I'll go through the code again as I clearly missed something. I thought I understood it.

Reply:
Okay,
So perhaps I'm wrong, but...

The HA mosquito user is set in the PW file
https://github.com/home-assistant/addons/blob/master/mosquitto/rootfs/etc/cont-init.d/mosquitto.sh#L33-L36 and doesn't authenticate from Mosquitto to the HA API.

And even if it did, Skipping Basic Auth would be fine as the Mosquitto Auth Plugin sends along a URL encoded string with the credentials that will get checked as long as the Authorization header is not present.
https://github.com/home-assistant/supervisor/blob/4416b6524eb120fc1456a305cf6dda49557c32cd/supervisor/api/auth.py#L59-L83

Reply:
It does: 

https://github.com/home-assistant/addons/blob/master/mosquitto/rootfs/usr/share/tempio/mosquitto.gtpl#L11-L30

Reply:
> It does:
> 
> https://github.com/home-assistant/addons/blob/master/mosquitto/rootfs/usr/share/tempio/mosquitto.gtpl#L11-L30

All this says is that it uses the HA HTTP API.
As far as I understand, That API can authenticate a user via Basic Auth, URL encoded form variables, or JSON body.

The Mosquitto Auth Plugin sends both Basic Auth and URL encoded form variables, when only one is required by HA Auth API
https://github.com/pvizeli/mosquitto-auth-plug/blob/master/src/be-http.c#L178-L197.

My change causes NGinx to remove the Authorization header from the request, thus bypassing the Basic Auth check and move onto authenticating the user via the  URL encoded form variables.

Reply:
But that would mean this would work around a Supervisor API issue? Or an issue in the Auth plugin? 

As in, Basic Auth versus URL encoded form variables should result in the exact same behavior?

Reply:
This is true. However Basic Auth only allows for a limited set of characters, whilst Mosquitto usernames and passwords are more diverse.

What would the impact of retiring Basic Auth from the Supervisor Auth API be?

I wonder if it's worth having the Auth Plugin control which method is sent to the backend HTTP server? @pvizeli thoughts?

Reply:
> What would the impact of retiring Basic Auth from the Supervisor Auth API be?

It is not about impact, it is about where this should be fixed. I feel like we are now working around a problem that shouldn't be "fixed" here, but on another place.

Reply:
Sorry perhaps I was unclear, I was suggesting that one option was perhapst that the Supervisory API no longer supports Basic Auth as a way or authorizing user credentials since it cannot handle the full set of characters supported by HA usernames and passwords.

Reply:
> I was suggesting that one option was perhapst that the Supervisory API no longer supports Basic Auth

It was introduced as the last of all authentication handling and actually used by almost all add-ons.

So thinking about the specifications and limitations, this is about the colon `:` I guess.
I suggest ensuring we don't accept colons in usernames. Reason why: This is making it work for this special case, but that doesn't solve the case that Home Assistant currently allows for it (apparently) and the Supervisor can't handle it (apparently).

TL;DR; this PR works around an issue, and is not address or fixing the greater problem behind it. Therefore, I don't think this PR should be merged.

Reply:
Disappointing conclusion, but thank you.